### PR TITLE
SFR-1672_ChicagoISACWebScraper

### PR DESCRIPTION
**Pull request by mitri-slory**
This is the current webscraper script for the Chicago ISAC catalogs https://isac.uchicago.edu/research/catalog-publications. The urls I commented out are the one with the the metadata I haven't fully scraped yet. Also, the deleteDuplicates script updates were just deduplicating the script rather than changing the logic and they're included in this branch since I thought it wasn't worth making a specific branch for those changes.

### CODE REVIEW

**Comment by sarangj** [comment]
I took a couple of looks at some of these links myself and it's really unfortunate that there aren't any helpful structures in the markup to make this simpler, props for destructuring all these pages.  I have a couple of thoughts for readability, but otherwise looks fine for this script!

File: BeautifulSoupScript.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+        #Testing to make sure pub is a <li> Tag and not an empty string
+        if not isinstance(pub, Tag):
+            continue
+
+        # if url == 'https://isac.uchicago.edu/research/publications/studies-ancient-oriental-civilization-saoc':
+        #     parsePubSAOC(pub, url)
+        #     continue
+
+        if pub.find('a', class_= "publication ss-standard ss-download btn"):
+            downloadLink = pub.find('a', class_= "publication ss-standard ss-download btn").get('href', None)
```
[the remark this answers is not in the document]
Comment (mitri-slory) [reply BeautifulSoupScript.py:101]: Nice catch. I decided to initialize a `downloadLinks` empty array in the beginning of the `parsePub2` method to resolve this issue.

File: BeautifulSoupScript.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+            finalMetaData.append({
+            'Volume': pub_tds[0].text,
+            'publicationInfo': pub_tds[2].text,
+            'isbnNumber': pub_tds[3].text,
+            'Extent': pub_tds[1].text,
+            'url': ''
+        })
+
+
+def parsePub2(elem, url):
```
[the remark this answers is not in the document]
Comment (mitri-slory) [reply BeautifulSoupScript.py:86]: I'll consider doing this as I web scrape more of the catalogs.

**Comment by sarangj** [comment]
looks good to me scanning the output json.  One last thing I'd suggest is to rename the script something more descriptive (sorry, missed that on the first pass!). Maybe like `chicago_isac_scraper.py` and maybe put it under the `scripts` directory?

**Review by sarangj — APPROVED** [review]

**Pull request merged**